cli/lockfile.rs

```diff
@@ -82,3 +82,158 @@ impl Lockfile {
     self.map.insert(specifier.to_string(), checksum);
   }
 }
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+  use std::fs::File;
+  use std::io::prelude::*;
+  use std::io::Write;
+  use tempfile::TempDir;
+
+  #[test]
+  fn new_nonexistent_lockfile() {
+    let file_path = String::from("nonexistent_lock_file.json");
+    assert!(Lockfile::new(file_path, false).is_err());
+  }
+
+  fn write_valid_lockfile(t: &TempDir) {
+    // create a valid lockfile for us to load
+    let file_path = t.path().join("valid_lockfile.json");
+    let mut file = File::create(file_path).expect("write file fail");
+    writeln!(file, "{{").expect("write line fail");
+    writeln!(file, "  \"https://deno.land/std@0.71.0/textproto/mod.ts\": \"3118d7a42c03c242c5a49c2ad91c8396110e14acca1324e7aaefd31a999b71a4\",").expect("write line fail");
+    writeln!(file, "  \"https://deno.land/std@0.71.0/async/delay.ts\": \"35957d585a6e3dd87706858fb1d6b551cb278271b03f52c5a2cb70e65e00c26a\"").expect("write line fail");
+    writeln!(file, "}}").expect("write line fail");
+  }
+
+  #[test]
+  fn new_valid_lockfile() {
+    let t = TempDir::new().expect("tempdir fail");
+    write_valid_lockfile(&t);
+
+    let file_path_buf = t.path().join("valid_lockfile.json");
```

Comment:
This line is duplicated in every test. Maybe replace `write_valid_lockfile` with `setup` that returns a) lockfile, b) tempdir c) lockfile path

Reply:
I have implemented a setup() and teardown() approach and it is looking a lot better.  I didn't end up returning the lockfile from setup because in at least 1 of the tests, the Lockfile constructor is part of the code that is 'under test' and also, I would have needed to parameterize the setup because some are constructed write=true and some write=false which feels a little labored.